File: Frontend/src/app/core/models/wishlist.ts
import { Product } from './product';

export interface WishlistItem {
  id: number;
  wishlistId: number;
  productId: string;
  product: Product;
  createdAt: string;
  updatedAt: string;
}

export interface Wishlist {
  id: number;
  userId: number;
  items: WishlistItem[];
  itemCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface AddToWishlistRequest {
  product_id: string;
}

File: Backend/app/models/wishlist.rb
class Wishlist < ApplicationRecord
  belongs_to :user
  has_many :wishlist_items, dependent: :destroy
  has_many :products, through: :wishlist_items

  validates :user_id, presence: true, uniqueness: true

  def item_count
    wishlist_items.count
  end

  def empty?
    wishlist_items.empty?
  end

  def clear_items
    wishlist_items.destroy_all
  end

  def as_json(options = {})
    {
      id: id,
      userId: user_id,
      items: wishlist_items.map(&:as_json),
      itemCount: item_count,
      createdAt: created_at.iso8601,
      updatedAt: updated_at.iso8601
    }
  end
end
